tests/streaming/action/test_dtmf.py

```diff
@@ -59,12 +70,74 @@ async def test_vonage_dtmf_press_digits(mocker, mock_env):
         assert action_output.response.success is True
 
 
+@pytest.fixture
+def mock_twilio_output_device(mocker: MockerFixture):
+    output_device = TwilioOutputDevice()
+    output_device.ws = mocker.AsyncMock()
+    output_device.stream_sid = "stream_sid"
+    return output_device
+
+
+@pytest.fixture
+def mock_twilio_phone_conversation(
+    mocker: MockerFixture, mock_twilio_output_device: TwilioOutputDevice
+):
+    twilio_phone_conversation_mock = mocker.MagicMock()
+    twilio_phone_conversation_mock.output_device = mock_twilio_output_device
+    return twilio_phone_conversation_mock
+
+
 @pytest.mark.asyncio
-async def test_twilio_dtmf_press_digits(mocker, mock_env):
+async def test_twilio_dtmf_press_digits(
+    mocker, mock_env, mock_twilio_phone_conversation, mock_twilio_output_device: TwilioOutputDevice
+):
     action = TwilioDTMF(action_config=DTMFVocodeActionConfig())
     digits = "1234"
     twilio_sid = "twilio_sid"
 
+    action.attach_conversation_state_manager(
+        TwilioPhoneConversationStateManager(mock_twilio_phone_conversation)
+    )
+
+    action_output = await action.run(
+        action_input=TwilioPhoneConversationActionInput(
+            action_config=DTMFVocodeActionConfig(),
+            conversation_id=create_conversation_id(),
+            params=DTMFParameters(buttons=digits),
+            twilio_sid=twilio_sid,
+        )
+    )
+
+    mock_twilio_output_device.start()
+    while not mock_twilio_output_device._twilio_events_queue.empty():
```

Comment:
Any situation where the queue could remain empty and this causes an infinite loop?

Reply:
good point, updated the while loop